### ISSUE TITLE
New tab proposal 

### ISSUE THREAD

**Issue by chblw** [dot-layer/blog#14]
Since a lot of the community is self-taught, I thought of a new tab to add at the top of the website. Parallel to the blog, we could expand on @davebulaval [bookshelf](https://github.com/dot-layer/community-events/blob/master/presentations/2018/chalet/bookshelf.md) where we could create hyperlinks to individual books reviews that members could add. 

The reviews would share a similar format and could quickly show topics addressed, difficulty, overall quality, etc. Then, a recap/highlights of the book. Then, members could discuss their experience with the book or ask questions. 

This could lead to different blog posts where if we find an interesting example/illustration/lab, we could make a blogpost and then refer to the book review. 

What do you guys think ?

**Comment by AntoineButeau** [comment]
We can just merge David bookshelf and add on top of it. 
How would the tab be named - Resources?

**Comment by chblw** [comment]
The bookshelf that David presented at the chalet. We would have a list of resources and once somebody reviews a book, we add a hyperlink to the review. We would also expand the list, the David bookshelf is just a starting point for resources.

**Comment by chblw** [comment]
List of possible names : 

- Resources
- Bookshelf
- Recommended reading
- Book reviews

**Comment by davebulaval** [comment]
Bookshelf reviews ?

**Comment by davebulaval** [comment]
The idea would a page with all book in category which the name would a link to an other post of the bookreview. 1 book == 1 book review. And let people at comment to the post. We want an introduction who can get be read in 30 sec and decide if you like it or not. It would be between 2-5 min of reading.

It will have tag, level of difficulty, and other.

**Comment by AntoineButeau** [comment]
We can either do a static page with one giant post or a series of posts that are tagged "bookshelf" and they show only on the "bookshelf" page.

Antoine Buteau

Engagement Manager
581-997-3658 ( tel:+15819973658 ) | LinkedIn ( https://www.linkedin.com/in/antoinebuteau/ )

**Comment by stecaron** [comment]
Here is how I would imagine it :

1. Let's have a static page (accessible from one tab that may be called "Bookshelf"). In this page, we'll just have the links to the online books (if they are open) and also have a link to a post that makes a review of this book (if we ever did one).

2. Let's create a category of post called "Book review" where we make a review of a book or multiple books. We could insert a link to the bookshelf in those posts as well.

That way, we'll have a straightforward recap page with all the links only. We could easily break down this page by subject and complexity just like @davebulaval did in his markdown. 
Also, we'll be able to see from the category tags, all the posts that are making reviews of books.

Make sense?

**Comment by chblw** [comment]
Yup that's how I saw it. We would start from David's bookshelf but remove the color indicators (keep that for the individual  book review). Also thought of having a template for reviews so they're easy to compare.

**Comment by davebulaval** [comment]
I think this is the best solution. My initial vision was that.

**Comment by davebulaval** [comment]
We also need a sort of factory of tag for the book. I don't want book to be tagged like this
BOOK A -> coding skill
BOOK B -> coding skil
BOOK C -> skill coding

**Comment by samperochkin** [comment]
J'ai push quelques modifs et ajouté un book. Je vais mettre un link pour un book dans un (new) post. Si je veux aussi mettre un lien vers le bookshelf: je fais juste mettre l'url comme si c'était n'importe quel autre site? (mettons que le bookshelf est sur master...)

**Comment by chblw** [comment]
Pour ajouter un book review, il faut créer un md dans content/bookshelf/**slug**. Ensuite, dans le hyperlink du livre dans le bookshelf, il faut faire un lien vers bookshelf/**slug**. 

Pour partager le bookshelf avec quelqu'un, le lien dot-layer.org/bookshelf va diriger vers la page.

Je travaille encore sur un template général pour faire des reviews.

**Comment by stecaron** [comment]
btw la page bookshelf est vraiment cool @chblw ! J'adore l'idée ! Ça serait bien de merger ça very soon, c'est du contenu nice pour le site !

**Comment by chblw** [comment]
Si je met post, ça affiche avec les tags. Si je mets page, ça n'affiche pas. Ça reste à décider si 
1. On le met en post ou en page
2. On met des catégories pour les voir sur le côté
3. On affiche les reviews sur la page principale quand on les publie

@stecaron @samperochkin @davebulaval vous pensez quoi des points 2 et 3 ?

**Comment by stecaron** [comment]
Perso je ne vois pas de problème à ce que les book reviews soient affichés sur la page principale comme les autres articles. Pour moi, un book review c'est un article tout simplement. Ça serait juste important qu'ils aillent tous la catégorie "Book review" comme ça on peut facilement voir les différents book reviews en cliquant sur la catégorie sur la sidebar. Et aussi les book reviews seront également accessible via le "Bookshelf" avec les liens.

Make sense @chblw ?

**Comment by davebulaval** [comment]
Je suis d'accord. C'était aussi mon idée.

**Comment by samperochkin** [comment]
Same same. Mais: dans ce cas @chblw , ça devient moins intéressant de faire des mini reviews comme tu m'avais demandé de faire (non?). J'ai l'impression que ça va polluer la page principale...

On serait capable d'avoir le choix? Mettons les review en profondeur on les affiche sur la page principale et les reviews ben basic on les affiche pas.

Un quick fix très laid serait de mettre une date antérieur (ou pas de date?) au book reviews qu'on veut pas au top de la liste des post sur la page principale. Si mettre pas de date ça marche ce serait pas si mal, sinon (mettre une date antérieur par exprès) c'est très deg comme façon de faire.

**Comment by chblw** [comment]
@stecaron @davebulaval Parfait je vais faire ça. @samperochkin On peut mettre "draft" dans l'entête et ça ne va pas apparaître sur la page principale je crois. À confirmer.

**Comment by stecaron** [comment]
@chblw faudrait tester si en mettant "draft" c'est quand même possible d'y accéder en cliquant sur un lien direct (comme un lien dans le book shelf par exemple).

**Comment by AnnieDeshaies** [comment]
@AntoineButeau , @caronl , @davebulaval , @samperochkin , @stecaron , @chblw est-ce qu'on veut encore faire ça? Je trouvais que c'est une bonne idée et la branche dort depuis 2018, sad. Je pourrais la dupliquer et vous proposez des idées de design pour que ça soit plus facile à naviguer. La liste des livres trigger mon âme de designer haha

**Comment by stecaron** [comment]
Je serais down qu’on remette ça en place je pense c’était une bonne idée a la place 😀

**Comment by AnnieDeshaies** [comment]
J'aime bien l'idée. Je repars de la branche bookshelf et on essaiera de faire avancer ça. Je vais essayer des designs et mettre les screenshots ici pour voir ce qui feel le mieux.

**Comment by davebulaval** [comment]
Je suis d'accord!

**Comment by guillaume-chevalier** [comment]
Shotgun Clean Code :smile: et j'y linkerai depuis mes 2 premiers articles. Sinon même, Clean Code et Clean Coder si vous préférez.

**Comment by AnnieDeshaies** [comment]
Je me suis faite une branche en locale et je suis repartie de bookshelf qui existait deja. J'aimerais essayer un design qui ressemble aux posts, mais un peu different pour les book reviews. C'est en WIP pour l'instant, je dois comprendre le framework en meme temps, je n'ai jamais travaille avec Hugo. Ca va bien par contre. J'affiche les reviews ¯\_(ツ)_/¯ s'agit juste de trouver un template qui inclut des headers significatifs et un look interessant pour la liste des previews et les post en single page.

**Comment by davebulaval** [comment]
Bookshelf: ajout recommandation livre en lien avec OpenLayer.
